rgwadmin/radosgw.go

```diff
@@ -0,0 +1,150 @@
+package rgwadmin
+
+import (
+	"bytes"
+	"context"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	v4 "github.com/aws/aws-sdk-go/aws/signer/v4"
+)
+
+const (
+	authRegion = "default"
+	service    = "s3"
+)
+
+var (
+	errNoEndpoint  = errors.New("endpoint not set")
+	errNoAccessKey = errors.New("access key not set")
+	errNoSecretKey = errors.New("secret key not set")
+)
+
+//API struct for New Client
+type API struct {
+	AccessKey  string
+	SecretKey  string
+	Endpoint   string
+	HTTPClient *http.Client
+	Debug      bool
+}
+
+// New returns client for Ceph RGW
+func New(endpoint, accessKey, secretKey string, httpClient *http.Client) (*API, error) {
+	// validate endpoint
+	if endpoint == "" {
+		return nil, errNoEndpoint
+	}
+
+	// validate access key
+	if accessKey == "" {
+		return nil, errNoAccessKey
+	}
+
+	// validate secret key
+	if secretKey == "" {
+		return nil, errNoSecretKey
+	}
+
+	// set http client, if TLS is desired it will have to be passed with an http client
+	if httpClient == nil {
+		httpClient = &http.Client{}
```

Comment:
how about adding a `timeout` for http request?

Reply:
I went with a 3sec timeout by default. Again configurable if a different client is passed.